### ISSUE TITLE
Getting nteract to find xeus-cling requires additional work 

### ISSUE THREAD

**Issue by jzf2101** [QuantStack/xeus-cling#176]
@rgbkrk and I are trying to get xeus on the nteract website but the way xeus installs intself doesn't allow us to easily get the kernels picked up by nteract.  one has to activate the environment and then call:

python -m ipykernel install --user --name cling --display-name "C++ (xeus-cling)" 

to get it to work.  Is there a way we can automate this?

**Comment by SylvainCorlay** [comment]
FYI the xeus-cling kernel installs its kernel spec into installation prefix/environment.

**Comment by rgbkrk** [comment]
Sadly, at least for all node driven frontends, we are not able to see conda environments because we're not launched as a python process. If a kernel.json can (optionally) be installed into e.g. `~/Library/Jupyter/kernels/cling/kernel.json`, even if it's referencing that conda environment, then we can detect it. Because of the nature of native applications, especially on mac, this is a bit of a hard requirement for us.

This all isn't to say we wouldn't ❤️ to be able to detect kernels in conda environments -- we got some pushback on auto-detecting conda environments on desktop.

**Comment by SylvainCorlay** [comment]
Yeah, i was just mentioning it in case you were running the server in the same environment (for a local run / trying out the kernel).

So you are stuck with the installation of the kernel spec...

**Comment by SylvainCorlay** [comment]
There is a kernel.json installed with the package, including an absolute path to the kernel (relocated by conda).

Also, if you are building from source, we have a cmake option for installing the kernel json in an extra location.

**Comment by jzf2101** [comment]
I think my question was that the kernel.json isn't something nteract can see since the enviornment has to be activated first to get access to it.  That's why I've written the additional ipykernel install line after it
